core/templates/services/keyboard-shortcut.service.spec.ts

```diff
@@ -0,0 +1,128 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the keyboard shortcut service.
+ */
+import 'mousetrap';
+
+import { KeyboardShortcutService } from 'services/keyboard-shortcut.service';
+
+describe('KeyboardShortcutService', () => {
+  var skipButton = document.createElement('button');
+  var nextButton = document.createElement('button');
+  var continueToNextCardButton = document.createElement('button');
+  var continueButton = document.createElement('button');
+  var backButton = document.createElement('button');
+  var searchBar = document.createElement('input');
+  var categoryBar = document.createElement('select');
+
+  const keyboardShortcutService = new KeyboardShortcutService();
+
+  beforeAll(() => {
+    skipButton.setAttribute('id', 'skipToMainContentId');
+    backButton.setAttribute('id', 'backButtonId');
+    nextButton.setAttribute('class', 'protractor-test-next-button');
+    continueButton.setAttribute('class', 'protractor-test-continue-button');
+    continueToNextCardButton.setAttribute(
+      'class', 'protractor-test-continue-to-next-card-button');
+    searchBar.setAttribute(
+      'class', 'protractor-test-search-input');
+    categoryBar.setAttribute('id', 'categoryBar');
+    document.body.append(skipButton);
+    document.body.append(continueButton);
+    document.body.append(backButton);
+    document.body.append(searchBar);
+    document.body.append(categoryBar);
+  });
+
+  it('should test navigation shortcuts.', () => {
+    var getStartedShortcutTriggered = false;
+    var libraryShortcutTriggered = false;
+    var learnerShortcutTriggered = false;
+    var creatorShortcutTriggered = false;
+    var preferenceShortcutTriggered = false;
+    var aboutShortcutTriggered = false;
+    var notificationShortcutTriggered = false;
+
+    keyboardShortcutService.setHref('#foo');
+    const reloadSpy = jasmine.createSpy('reload');
+    spyOn(keyboardShortcutService, 'setHref').and.callFake(function(href) {
+      if (href === '/get-started') {
+        getStartedShortcutTriggered = true;
+      } else if (href === '/community-library') {
+        learnerShortcutTriggered = true;
+      } else if (href === '/learner-dashboard') {
+        libraryShortcutTriggered = true;
+      } else if (href === '/creator-dashboard') {
+        creatorShortcutTriggered = true;
+      } else if (href === '/preferences') {
+        preferenceShortcutTriggered = true;
+      } else if (href === '/about') {
+        aboutShortcutTriggered = true;
+      } else if (href === '/notifications') {
+        notificationShortcutTriggered = true;
+      }
+    });
```

Comment:
Instead of setting boolean flags, what if we just record the href argument string in a variable. Then the test would look like the following:

```
...

Mousetrap.trigger('ctrl+0');
expect(href).toEqual('/get-started');

Mousetrap.trigger('ctrl+1');
expect(href).toEqual('/community-library');

...
```

Reply:
done